### PR TITLE
Conventionize

### PR DESCRIPTION
**Pull request by oneacik**
What this PR does?:
- Ran prettier on this code - please suggest another approach if you want to preserve the formatting.
- Encapsulated the pluginManager, so it handles commands by itself.
- Core now has only plugging pluginManager to discord api responsibility.
- Removed cursed plugin finding capability for injection by constructor argument

Why?:
I want in the future to create some plugin for SpottyKiwi and I am doing some cleaning so it will be easier to manipulate it.

### CODE REVIEW

File: lib/plugins/pluginManager.ts [hunk 1]
```diff
@@ -1,78 +1,124 @@
...
+        try {
+            logger.info("Handling message.", logContext);
+            foundMatchingCommand.handle(interaction);
         } catch (error) {
-            logger.error(error)
+            logger.error("An error occurred.", logContext, { error });
         }
     }
 
+    public async handleMessage(interaction: Message) {
```
Comment (oneacik) [inline lib/plugins/pluginManager.ts:92]: There.

File: lib/core.ts [hunk 2]
```diff
@@ -1,72 +1,32 @@
...
-
-    const client = new Client({ intents: [Intents.FLAGS.GUILDS, Intents.FLAGS.DIRECT_MESSAGES, Intents.FLAGS.GUILD_MESSAGES] })
-    const pluginManager = new PluginManager()
-    pluginManager.loadPlugins()
-
-    client.on('ready', () => {
-        logger.info('Bot is ready.')
-    })
-
-    client.on('interactionCreate', async interaction => {
```
Comment (oneacik) [inline lib/core.ts:19]: Moved feature envy from Here to [There](https://github.com/hspsh/spotty-kiwi/pull/39/files#r775252413).

**Comment by oneacik** [comment]
How To check this PR?:

Checking this code by going file by file may injure you.
Please take a look on this commit: https://github.com/hspsh/spotty-kiwi/pull/39/commits/f34d4fce0281af17ac7d068d36f25982c72afc76 
And then just take a holistic look on other changes.

**Comment by LiquidLemon** [comment]
Hey, good job, this looks pretty clean. I should've done this a long time ago to make it easier for people to contribute. I'm not a huge fan of having to register every plugin manually but I guess it's cleaner than that cursed discovery function.

As far as Prettier goes - I have no strong opinion one way or another but there is already an ESLint config in this repo. We can ditch that for Prettier but then please delete the old config so it's not misleading. It would be amazing to have formatting enforced by CI (possibly also having it run automatically with pre-commit?) but if you don't feel like setting up GitHub actions then we can just leave that for the future.

File: lib/core.ts (hunk 2 above)
Comment (LiquidLemon) [reply lib/core.ts:19]: Sure, I didn't plan on this staying here forever.

**Comment by oneacik** [comment]
> Hey, good job, this looks pretty clean. I should've done this a long time ago to make it easier for people to contribute. I'm not a huge fan of having to register every plugin manually but I guess it's cleaner than that cursed discovery function.
> 
> As far as Prettier goes - I have no strong opinion one way or another but there is already an ESLint config in this repo. We can ditch that for Prettier but then please delete the old config so it's not misleading. It would be amazing to have formatting enforced by CI (possibly also having it run automatically with pre-commit?) but if you don't feel like setting up GitHub actions then we can just leave that for the future.

I will now try to integrate prettier with eslint or see if eslint isn't already addressing formatting concern.
I will leave pipeline for later.

**Comment by oneacik** [comment]
> Hey, good job, this looks pretty clean. I should've done this a long time ago to make it easier for people to contribute. I'm not a huge fan of having to register every plugin manually but I guess it's cleaner than that cursed discovery function.
> 
> As far as Prettier goes - I have no strong opinion one way or another but there is already an ESLint config in this repo. We can ditch that for Prettier but then please delete the old config so it's not misleading. It would be amazing to have formatting enforced by CI (possibly also having it run automatically with pre-commit?) but if you don't feel like setting up GitHub actions then we can just leave that for the future.

I made eslint and prettier not conflicting with each other.
As prettier is for style and eslint is for static analysis mostly.
This PR now should be ready to check and hopefully merge.

**Comment by LiquidLemon** [comment]
Okay, looks good now, let's go.

**Review by LiquidLemon — APPROVED** [review]

**Pull request merged**